### DISCUSSION TITLE
Authentication Best Practices

### DISCUSSION THREAD

**Discussion by khaosyy** [pocketbase/pocketbase#2397] [Q&A]
Hi, not sure if this is right.

I set my view auth settings to

```
List/Search Rule: @request.auth.id = uid 
```
I'm not sure if this qualifies as securing the collections/documents. I would imagine a hacker could manually change their auth.id to the uid that they want, if they had a list of user uids?

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. What exactly is the use case and the desired outcome?

`@request.auth.id` can't be changed. It's the id of the authenticated auth record.
It is possible to explicitly set the `id` only during the creation of a record and it can't be modified afterwards via the web APIs.

If you haven't already, you can find some info about the collection rules in https://pocketbase.io/docs/api-rules-and-filters/.

**Reply by navaneeth-dev** [reply]
@request.auth.id is got from JWT token which is created and singed by server with secret AFAIK. So no an attacker cannot just change id
